src/eth/primitives/slot.rs

```diff
@@ -202,3 +204,11 @@ impl PgHasArrayType for SlotValue {
         <[u8; 32] as PgHasArrayType>::array_type_info()
     }
 }
+
+#[derive(Debug, sqlx::Decode)]
+pub struct SlotSample {
+    pub address: Address,
+    pub block_number: BlockNumber,
+    pub index: SlotIndex,
```

Comment:
Can you use `Slot` directly here instead of `index` and `value`?

Reply:
That what I attempted at first, but I couldn't make `query_file_as!` play nicely with nested structs. This is a known issue https://github.com/launchbadge/sqlx/discussions/1446. I could use the query function and then map the result. But I found adding the index and value as fields so that I could use `query_file_as!` nicer.